indexer/services/socks/src/websocket/index.ts

```diff
@@ -396,7 +382,6 @@ export class Index {
 
       // Delete subscription data.
       this.subscriptions.remove(connectionId);
-      this.pingHandler.handleDisconnect(connectionId);
```

Comment:
nit: Can you also remove the `onMessage` listener in `disconnect`, before the `terminate` call? If there is no heartbeat, we directly call `disconnect` rather than `ws.close` so we'd want to ensure there's no message listener on these connections as well.

Reply:
I don't think that's necessary. Terminate should destroy the websocket. https://stackoverflow.com/questions/74081816/when-is-it-preferable-to-use-ws-terminate-vs-ws-close

Reply:
Nvm added